### PR TITLE
feat: GSTR3B Report json creation and print format

### PR DESCRIPTION
**Pull request by deepeshgarg007**
Json Creation:
screenshot 2019-02-07 at 9 16 07 am
Print Format:
screenshot 2019-02-07 at 9 16 15 am

### CODE REVIEW

File: erpnext/regional/doctype/gstr_3b_report/gstr_3b_report.html [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+		font-size: 8.0pt !important;
+		font-family: Tahoma, sans-serif;
+	}
+	.disabled {
+		background-color: #d9d9d9;
+	}
+
+</style>
+<h3 class="text-center">GSTR3B-Form</h3>
+<h5>GSTIN:&nbsp{{ data.gstin }}</h5>
```
Comment (sagarvora) [inline erpnext/regional/doctype/gstr_3b_report/gstr_3b_report.html]: One space can be added without using `&nbsp`.

File: erpnext/regional/doctype/gstr_3b_report/gstr_3b_report.html [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+
+<h5>3.1&nbsp&nbspDetails of Outward Supplies and inward supplies liable to reverse charge</h5>
```
Comment (sagarvora) [inline erpnext/regional/doctype/gstr_3b_report/gstr_3b_report.html]: Better to use `&nbsp;`

And have a space after it too, helps with readability.

File: erpnext/patches/v12_0/set_gst_category.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+		"input service": "Import Of Service",
+		"capital goods": "Import Of Capital Goods",
+		"input": "Input Service Distributor"
+	}
+
+	has_gst_fields = frappe.db.has_column('Purchase Invoice','eligibility_for_itc')
+
+	if has_gst_fields:
+		for old, new in itc_update_map.items():
+			frappe.db.sql("UPDATE `tabPurchase Invoice` SET eligibility_for_itc = replace(eligibility_for_itc, %s, %s)", (old, new))
```
Comment (karthikeyan5) [inline erpnext/patches/v12_0/set_gst_category.py]:
```suggestion
			frappe.db.sql("UPDATE `tabPurchase Invoice` SET eligibility_for_itc = %s where eligibility_for_itc = %s", (new, old))
```
this query would be faster to execute and is more correct in this use case.

**Comment by karthikeyan5** [comment]
since the `invoice_type` field was removed in the patch can you please update the same in the following reports:

- gst_purchase_register

**Review by karthikeyan5 — CHANGES_REQUESTED** [review]

**Comment by test2a** [comment]
yay...... finally after such a long time, GSTR 3B is finally here. Good job

**Review by karthikeyan5 — DISMISSED** [review]

**Review by netchampfaris — APPROVED** [review]

File: erpnext/regional/doctype/gstr_3b_report/gstr_3b_report.html [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+		font-size: 8.0pt !important;
+		font-family: Tahoma, sans-serif;
+	}
+	.disabled {
+		background-color: #d9d9d9;
+	}
+
+</style>
+<div>
+	<h3 class="text-center">GSTR3B-Form</h3>
```
Comment (netchampfaris) [inline erpnext/regional/doctype/gstr_3b_report/gstr_3b_report.html]: Add Translated Strings

**Review by netchampfaris — CHANGES_REQUESTED** [review]

**Review by netchampfaris — APPROVED** [review]

**Pull request merged**